### PR TITLE
Initialise plugins in frontend hook

### PR DESCRIPTION
**Pull request by mpickering**
The generic frontend already contains this call to do the initialisation
but because we bypassed it, the plugins were never being loaded.

There is probably more we can do to stop repeatedly loading some plugins
but until someone complains, plugins are quite rare so not going to
invest the effort.

### CODE REVIEW

**Comment by isovector** [comment]
Excellent, thanks!

File: src/HIE/Bios/Ghc/Load.hs [hunk 1]
```diff
@@ -114,12 +115,21 @@ collectASTs action = do
...
+  p <- G.parseModule =<< initializePluginsGhc ms
   tcm <- G.typecheckModule p
   let tcg_env = fst (tm_internals_ tcm)
   liftIO $ modifyIORef tc_ref (tcm :)
   return $ FrontendTypecheck tcg_env
 
+initializePluginsGhc :: ModSummary -> Ghc ModSummary
+initializePluginsGhc ms = do
+  hsc_env <- getSession
+  df <- liftIO $ initializePlugins hsc_env (ms_hspp_opts  ms)
```
Comment (fendor) [inline src/HIE/Bios/Ghc/Load.hs]: Needs to be guarded by CPP or defined in a Compat Module

**Comment by fendor** [comment]
Otherwise: the pr works as intended with Haskell-IDE-Engine!

**Comment by mpickering** [comment]
I fixed CI and squashed the commits together.

**Pull request merged**